src/olympia/amo/management/commands/check_locales_completion_rate.py

```diff
@@ -0,0 +1,175 @@
+import re
+
+import hashlib
+import requests
+from collections import defaultdict
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from django.template import loader
+
+from olympia.amo.utils import send_mail
+from olympia.core.languages import ALL_LANGUAGES
+
+
+class Command(BaseCommand):
+    help = (
+        'Check locales completion rate for AMO projects in Pontoon and warn '
+        'us by email of any falling below pre-established thresholds'
+    )
+    PONTOON_API = 'https://pontoon.mozilla.org/graphql'
+    PONTOON_PROJECTS = ('amo', 'amo-frontend')
+    QUERY_TEMPLATE = """
+        {
+          project(slug: "%(project)s") {
+            name,
+            localizations {
+              locale {
+                code,
+                name
+              }
+              totalStrings,
+              approvedStrings
+            }
+          }
+        }
+    """
+    # 40% completion is the first step of the project. Then we'll move it up to
+    # 80% for a limited period, and eventually it will be set to 70% to account
+    # for new strings being added over time.
+    COMPLETION_THRESHOLD = 40
+    # Arbitrary list of locales we always want to keep, either because they
+    # receive the most traffic/affect the most users, or because they are RTL
+    # and we want to keep at least one of those to ensure AMO works in RTL.
+    LOCALES_TO_ALWAYS_KEEP = (
+        'ar',
+        'de',
+        'en-US',
+        'es-ES',
+        'fr',
+        'he',
+        'it',
+        'ja',
+        'pl',
+    )
+    EMAIL_RECIPIENTS = ('amo-locales-notifications@mozilla.com',)
+    EMAIL_SUBJECT = 'AMO locales completion rate check'
+    EMAIL_TEMPLATE_PATH = 'amo/emails/locales_completion.ltxt'
+
+    def add_arguments(self, parser):
+        """Handle command arguments."""
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            default=False,
+            help='Fetch data from Pontoon but do not send emails.',
+        )
+
+    def handle(self, *args, **options):
```

Comment:
This is quite a large command and I wonder how it will actually run in the end if it is run via a cron job that executes the command.. do crons run on worker by default?

Wondering also because it might be nice to have some basic retry logic here in case of a temporary failure. using a task with basic retry could give us this out of the box.. just thoughts. I know this is kind of a "first step" but that decision is easier to make right the first time than to fix later.

Reply:
It's running fairly quickly (seconds on my machine) and it's not a critical command... crons run on a special dedicated instance.

I don't think it's worth having this as a task, but I could add retries support through `requests` if we think that's worth it ? Ultimately it's not a critical command, and we'll see failures in Sentry so I wasn't particularly worried about it.

Reply:
Added basic `requests` retrying (6 tries, with backoff)

Reply:
Since we recently talked about retries, I thought it a good opportunity to promote that as the _reflexive and default_ behavior for work that depends on external services. But ok, like you said it's not critical.